packages/oruga-next/src/components/input/Input.vue

```diff
@@ -396,6 +408,12 @@ const rootClasses = defineClasses(
         null,
         computed(() => props.expanded),
     ],
+    [
+        "hasIconRightClass",
```

Comment:
I don't think this class is really necessary, you can check with css if the right icon is enabled with something like ` input, textarea { & + .iconRightClass {...}}`.

Reply:
You're right that it could work like that, but this is another spot where this is just how Bulma works. [See this code](https://github.com/jgthms/bulma/blob/master/sass/form/tools.sass#L171) for details.

Reply:
Actually, now that I look at this again, I need to add a `hasIconLeft` prop as well.

Reply:
Maybe then a `hasIconClass` with the suffix like `left`, `right`,  `both` ?

Reply:
If both are present there needs to be two classes rather than one. The has icon left class is currently handled in the theme because it only needs prop data, while the right class needs access to the computed.

Reply:
but the theme could handle it with a function based on the given suffix? The function got called every time the suffix change. 

However, I'm also fine with adding two classed one for left and one for right, maybe its more clear.

Reply:
Maybe? I can't think of a way to do it with the existing suffix mechanism, but that doesn't mean it's impossible. Either way, I think we should keep it like this for now. It's clear and simple this way.